### DISCUSSION TITLE
How to update `gradsflow` on conda-forge, once a new PyPI version is released?

### DISCUSSION THREAD

**Discussion by sugatoray** [gradsflow/gradsflow#145] [Q&A] [answered]
What are the steps to update `gradsflow` on conda-forge once a new version on PyPI is released?

**Comment by sugatoray** [comment, marked as answer]
## Process to raise a PR for conda-forge package update

- **Step-1**: Go to: https://github.com/conda-forge/gradsflow-feedstock
- **Step-2**: Fork it. Create a new branch (for example: `update_v1.2.3` to update some library to it's version `1.2.3`).
- **Step-3**: Update the `recipe/meta.yaml`.
  - [x] Update the jinja2 variable: `version` (same as PyPI)
  - [x] Update the field: `sha256` (should be same as PyPI source, `<package>.tar.gz` file)
  - [x] If changing the version, DON'T UPDATE the build number. 
  - [x] If making a change to the recipe, without any version update, increase the build number by 1.
  - [x] Update the dependencies under run-requirements (these are the ones you typically include in `setup.py`/ `requirements.txt`).
    - [x] add newer dependencies (they must exist on conda-forge channel, if not speak with me)
    - [x] update version specification for each dependency
    - [x] use the more classical `>`, `=`, `<` for defining version constraints instead of `^`, `~`.
    - [x] drop dependencies that are no longer included in your `setup.py` 
    - [x] NOTE: optional dependencies should be dropped here, so users can install them only if they need them. Ideally, such optional dependencies should be made part of `extras` under `setup.py`.
- **Step-4**: Submit a PR (follow the instructions while opening the PR).
- **Step-5**: Review
  - [x] If all CI's have passed (`pytorch` is not available on conda-forge for windows, so, typically the windows CI run will fail, and that is okay), then we move to merging.
  - [x] A maintainer will review and once all looks good will merge it. 
     > @aniketmaurya Please let me review your feedstock PRs a couple of times initially, to help you ease into the process. 
- **Step-6**: Once merged, the new package will be available on the channel in 2-3 hours.

**Reply by aniketmaurya** [reply]
thank you @sugatoray for the detailed documentation 🙏🏻

**Reply by sugatoray** [reply]
In most cases, given your dependencies haven't changed, all you would need to change is `version` and `sha256`.
